lib/runtime/invalid_transaction.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package runtime
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// InvalidTransaction is child VDT of TransactionValidityError
+type InvalidTransaction scale.VaryingDataType
+
+// Index returns the VDT index
+func (InvalidTransaction) Index() uint {
+	return 0
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (i *InvalidTransaction) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*i)
+	err = vdt.Set(val)
+	if err != nil {
+		return err
+	}
+	*i = InvalidTransaction(vdt)
+	return nil
+}
+
+// Value will return value from underying VaryingDataType
+func (i *InvalidTransaction) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*i)
+	return vdt.Value()
+}
+
+// Error returns the error message associated with the InvalidTransaction
+func (i InvalidTransaction) Error() string {
+	value := i.Value()
+	if value == nil {
+		return "invalidTransaction hasn't been set"
+	}
+	err, ok := value.(error)
+	if !ok {
+		panic(fmt.Sprintf("%T does not implement the error type", value))
+	}
+	return err.Error()
+}
+
+// NewInvalidTransaction is constructor for InvalidTransaction
+func NewInvalidTransaction() InvalidTransaction {
+	vdt, err := scale.NewVaryingDataType(Call{}, Payment{}, Future{}, Stale{}, BadProof{}, AncientBirthBlock{},
+		ExhaustsResources{}, InvalidCustom(0), BadMandatory{}, MandatoryDispatch{})
+	if err != nil {
+		panic(err)
+	}
+	return InvalidTransaction(vdt)
+}
+
+// Call The call of the transaction is not expected
+type Call struct{}
+
+// Index returns the VDT index
+func (Call) Index() uint { return 0 }
+
+// Error returns the error message associated with the Call
+func (Call) Error() string {
+	return "call of the transaction is not expected"
+}
+
+// Payment General error to do with the inability to pay some fees (e.g. account balance too low)
+type Payment struct{}
```

Comment:
I believe they need to be exported to 1) be unmarshalled (since they are the outer type yes?) and also 2) to adhere to the VDT Value interface.  @timwu20 can verify this

Reply:
The attributes need to be exported when dealing with structs, not necessarily the struct itself.  Either way if these types are supposed to mimic exported (`pub`) types in substrate, I'd go ahead and export them as well.